crates/codec/src/context.rs

```diff
@@ -0,0 +1,41 @@
+//! Encoding of a binary [`Context`].
+//!
+//! ```text
+//!
+//!  +------------------+-----------------+-----------------+
+//!  |                  |                 |                 |
+//!  |  Transaction Id  |  Current Layer  |  Current State  |
+//!  |     (Hash)       |     (u64)       |     (State)     |
+//!  |                  |                 |                 |
+//!  |    32 bytes      |    8 bytes      |    32 bytes     |
+//!  |                  |   (Big-Endian)  |                 |
+//!  |                  |                 |                 |
+//!  +------------------+-----------------+-----------------+
+//!
+//! ```
+
+use std::io::Cursor;
+
+use svm_types::{Context, Layer};
+
+use crate::{ReadExt, WriteExt};
+
+/// Encodes a binary [`Context`] of a transaction.
+pub fn encode(context: &Context, w: &mut Vec<u8>) {
+    w.write_tx_id(context.tx_id());
+    w.write_u64_be(context.layer().0);
+    w.write_state(context.state());
+}
+
+/// Decodes a binary [`Context`] of a transaction.
+///
+/// Returns the decoded [`Context`],
+/// On failure, returns [`std::io::Result`].
+pub fn decode(cursor: &mut Cursor<&[u8]>) -> std::io::Result<Context> {
+    let tx_id = cursor.read_tx_id()?;
+    let layer = cursor.read_u64_be()?;
+    let state = cursor.read_state()?;
+
+    let context = Context::new(tx_id, Layer(layer), state);
+    Ok(context)
+}
```

Comment:
As I've said, I'd really prefer to keep the `types` from encoding in isolation for now.

A good example from the current code base is the `svm-storage` crate.
It's using the `svm-types` as a depedendency.
However, It doesn't rely on the `svm-codec` (since it's Transaction & Receipts focused).

Similarly, the new Global-State will probably use `svm-types` and it'll not have any reason I can currently imagine 
to use the `svm-codec`.

The `svm-codec` is more for communication with the external (out of SVM) world.
It might be Transactions, Receipts or some `runtime-ffi` (In-memory buffers encoding).